### PR TITLE
[DEV-3478] Add CFDA number to transactions endpoint

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Adds the CFDA number for the transaction to the api/v2/transactions endpoint for assistance transactions to support the addition of the CFDA to the  
**Technical details:**
Annotates the TransactionNormalized query with a subquery to TransactionFABS.

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend
4. [N/A] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-3478](https://federal-spending-transparency.atlassian.net/browse/DEV-3478):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No change to matviews, no ops tickets needed
```

### CODE REVIEW

File: usaspending_api/awards/v2/views/transactions.py [hunk 1]
```diff
@@ -41,9 +41,9 @@ class TransactionViewSet(APIView):
     }
 
     def __init__(self):
-        models = customize_pagination_with_sort_columns(
-            list(TransactionViewSet.transaction_lookup.keys()), "action_date"
-        )
+        sort_cols = list(TransactionViewSet.transaction_lookup.keys())
+        sort_cols.append("cfda_number")
```
Comment (sethstoudenmier) [inline usaspending_api/awards/v2/views/transactions.py]: Could you just add `cfda_number` to the `transaction_lookup` above?

Comment (sethstoudenmier) [reply usaspending_api/awards/v2/views/transactions.py]: Nevermind, just noticed the bit below.

File: usaspending_api/awards/v2/views/transactions.py [hunk 2]
```diff
@@ -85,6 +95,8 @@ def _format_results(self, rows):
             result = {k: row[v] for k, v in self.transaction_lookup.items() if k != "award_id"}
             if result["is_fpds"]:
                 unique_prefix = "CONT_TX"
+            else:
+                result["cfda_number"] = row["cfda_number"]
```
Comment (sethstoudenmier) [inline usaspending_api/awards/v2/views/transactions.py]: Not worth blocking this PR on, but if you do another round of changes at all might be good to move `unique_prefix = "ASST_TX"` into this else to keep all FABS specific parts together.

File: usaspending_api/awards/v2/views/transactions.py [hunk 3]
```diff
@@ -68,8 +68,18 @@ def _business_logic(self, request_data: dict) -> list:
         lower_limit = (request_data["page"] - 1) * request_data["limit"]
         upper_limit = request_data["page"] * request_data["limit"]
 
-        queryset = TransactionNormalized.objects.all().values(*list(self.transaction_lookup.values())).filter(**filter)
-
+        queryset = (
+            TransactionNormalized.objects.all()
+            .values(*list(self.transaction_lookup.values()))
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/views/transactions.py]: I haven't tried it, might be cleaner to use `.select_related('transaction_fabs')` as long as it translated to a left join

Comment (emily-brents) [reply usaspending_api/awards/v2/views/transactions.py]: good call, it makes it much cleaner.

**Pull request merged**